### ISSUE TITLE
Improve IDE comfort for DSL usage

### ISSUE THREAD

**Issue by jcornaz** [rjaros/kvision#93]
Hi, 

Trying to write the UI using the DSL builders (as mentioned here: https://kvision.gitbook.io/kvision-guide/part-1-fundamentals/ui-structure#dsl), the IDE (in my case IntelliJ) is not helping, because no auto-completion is showing up.

![image](https://user-images.githubusercontent.com/2535178/66866415-89a77a80-ef99-11e9-94c3-a322759382bb.png)

Looking in the source code, I realize that it is because all the extension functions for DSL builder are written in companion objects:
```kotlin
class HPanel {
    companion object {
        fun Container.hPanel(...): HPanel { ... }
    }
}
```

So, despite it is possible to use the DSL builders as shown in the doc, it needs imports such as:

```kotlin
import pl.treksoft.kvision.panel.HPanel.Companion.hPanel
```

And IntelliJ does not seem to be able to propose auto-completion for it.

I propose one of the following options:
1. Don't write theses function in a companion object*
2. Document how to setup IntelliJ so that we can have good auto-completion
3. Publish an IntelliJ plugin

*I have to say, I don't understand the reasoning behind putting them in the companion object, and I'd be interested to learn the rationale behind it.

**Comment by rjaros** [comment]
From my experience it's enough to write `hPanel()` (with brackets), and the auto-completion popup will appear. Have you tried that?

**Comment by jcornaz** [comment]
Yes, we have quick fix to import. But it is too late. By definition, I want auto-**completion** before I am done writing it.

**Comment by jcornaz** [comment]
It causes me two concrete problems:

1. I have a hard time to discover the API. My only option is to read all the doc. Which kills a big benefit of statically typed languages: Being able to discover the API directly when using it, leveraging auto-completion.

2. I am so used to get auto-completion for everything and using it for 99% of the things I type, that if I'm halfway writing a word (let's say `hPan` for `hPanel()`) and no auto-completion shows up, I get convinced that I am doing it wrong. I don't even complete the word, and I directly try something different and go the documentation.

**Comment by rjaros** [comment]
It's now or never. The 2.0 move is the only moment to do such breaking change. Let's try.

**Comment by rjaros** [comment]
Released with 2.0.0-M3.
